lib/index.js

```diff
@@ -11,79 +11,78 @@ const Hoek = require('hoek');
 const internals = {};
 
 
-exports.register = function (plugin, options, next) {
+exports.plugin = {
+    pkg: require('../package.json'),
+    register: function (server) {
 
-    plugin.auth.scheme('basic', internals.implementation);
-    next();
-};
-
-
-exports.register.attributes = {
-    pkg: require('../package.json')
+        server.auth.scheme('basic', internals.implementation);
+    }
 };
 
 
 internals.implementation = function (server, options) {
 
     Hoek.assert(options, 'Missing basic auth strategy options');
-    Hoek.assert(typeof options.validateFunc === 'function', 'options.validateFunc must be a valid function in basic scheme');
+    Hoek.assert(typeof options.validate === 'function', 'options.validate must be a valid function in basic scheme');
 
     const settings = Hoek.clone(options);
 
     const scheme = {
-        authenticate: function (request, reply) {
+        authenticate: async function (request, h) {
+
+            const authorization = request.headers.authorization;
 
-            const req = request.raw.req;
-            const authorization = req.headers.authorization;
             if (!authorization) {
-                return reply(Boom.unauthorized(null, 'Basic', settings.unauthorizedAttributes));
+                return h.unauthenticated(Boom.unauthorized(null, 'Basic', settings.unauthorizedAttributes));
```

Comment:
If you are not also passing credentials, you can just throw the error.

Reply:
👍